ambry-tools/src/main/java/com.github.ambry/tools/perf/rest/NettyPerfClient.java

```diff
@@ -533,10 +590,17 @@ public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) {
     private void sendRequest(ChannelHandlerContext ctx) {
       requestId++;
       long globalId = totalRequestCount.incrementAndGet();
-      logger.trace("Sending request with global ID {} and local ID {} on channel {}", globalId, requestId,
-          ctx.channel());
+      Channel channel = ctx.channel();
+      String hostname = ((SocketChannel) channel).remoteAddress().getHostName();
+      hostToRequestCount.get(hostname).incrementAndGet();
+      logger.trace("Sending request with global ID {} and local ID {} on channel {}", globalId, requestId, channel);
       reset();
       perfClientMetrics.requestRate.mark();
+      try {
+        Thread.sleep(hostToSleepTime.get(hostname));
```

Comment:
nit: Since this function will be called within a nonblocking function, maybe we shouldn't do a thread.sleep here.
Some think like
```
Channel channel = ctx.Channel();
ctx.channel().eventLoop().schedule(new Runnable() {
    // send the request out.
}, hostToSleepTime.get(hostname), TimeUnit.MILLSECOND);
```

Reply:
Thanks for letting me know this. Temporarily use current version, will take your suggestion in future PR after several tests.